### PR TITLE
Use buildpackage

### PR DESCRIPTION
**Pull request by rikhuijzer**
I noticed that we can replace 
```
using Pkg; Pkg.activate("."); Pkg.instantiate();
```
by just using the `--project` flag if we also run 
```
uses: julia-actions/julia-buildpkg@master
```
Also, I've capitalized the filename `deploy.yml` because that appears to be the convention in Julia. For example, see https://github.com/JuliaRegistries/CompatHelper.jl.

### CODE REVIEW

**Comment by rikhuijzer** [comment]
The reason that I suggest this is because we can then move some complexity out of the CI file and into `julia-buildpkg`. For example, they currently handle newly registered packages which do not show up at pkg.julialang.org yet, see https://github.com/julia-actions/julia-buildpkg/blob/master/action.yml. Also, it is more consistent with the build in other packages, so I think this could be more clear to new and semi-new users because
```
using Pkg; Pkg.activate("."); Pkg.instantiate();
```
requires a bit of Julia knowledge to understand what is going on, whereas
```
uses: julia-actions/julia-buildpkg@master
```
clearly is something with "building" the "pkg", which I guess, makes sense to people.

**Comment by tlienart** [comment]
Hey Rik, thanks a lot for this, sorry I haven't had the time to check this yet. Could you also point me to a repo where you used this? Thanks!

**Comment by rikhuijzer** [comment]
I tested it on a private repository. Now, I've also used it at https://github.com/StatisticalRethinkingJulia/TuringModels.jl/pull/60. The simplify workflow commit in that PR changes the CI according to this PR. As you can see, CI was green before and after that change.

File: src/templates/common/.github/workflows/Deploy.yml [hunk 1]
```diff
@@ -24,18 +24,19 @@ jobs:
       uses: julia-actions/setup-julia@v1
       with:
         version: 1.5
+    - name: Install dependencies listed in Project.toml and build package
+      uses: julia-actions/julia-buildpkg@master
     # NOTE
-    #   The steps below ensure that NodeJS and Franklin are loaded then it
-    #   installs highlight.js which is needed for the prerendering step
+    #   Using --project below ensures that NodeJS and Franklin are loaded then 
+    #   it installs highlight.js which is needed for the prerendering step
```
Comment (tlienart) [inline src/templates/common/.github/workflows/Deploy.yml:31]: This comment (re NodeJS and Franklin) is now incorrect right? what it means is that it'll ensure that whatever you have in your Project.toml is installed, correct?

Comment (rikhuijzer) [reply src/templates/common/.github/workflows/Deploy.yml:31]: The word "activated" might be better indeed. `julia --project` is identical to `julia -e 'using Pkg; Pkg.activate(".")` as far as I know.

Comment (tlienart) [reply src/templates/common/.github/workflows/Deploy.yml]: ok cool, I'll merge this and do light edits to reflect this

**Comment by rikhuijzer** [comment]
That's a quick merge after the changes :yum: Thanks :+1:

**Comment by tlienart** [comment]
Thanks Rik! I did some light additional edits but I think this is better.

By the way, I'm re-testing this as the CI doesn't indicate whether this specific scripts work (the deploy is different)

**Pull request merged**